### PR TITLE
Init adaptation for Pytorch 0.4.0

### PR DESCRIPTION
**Pull request by zhengzx-nlp**
shared mode does not work yet

### CODE REVIEW

**Comment by whr94621** [comment]
The only thing is should we use ```enable_grad()```, ```set_grad_enabled()```, ```no_grad()``` as a context-manager.
I think a context-manager is more clear that the gradient mode is only defined in such a block below ```with``` statement.

**Comment by whr94621** [comment]
I will merge this PR first. We can complete adaptation afterwards :)

File: src/main.py [hunk 1]
```diff
@@ -38,22 +47,25 @@ def _np_pad_batch_2D(samples, pad, batch_first=True, volatile=False, cuda=True):
...
-        x = Variable(torch.LongTensor(x_np).contiguous(),
-                     volatile=volatile)
+        x = torch.tensor(x_np)
+
         if cuda is True:
             x = x.cuda()
         return x
 
+    grad_mode = not eval  # True for training, False for eval/infer
+    torch.set_grad_enabled(grad_mode)  # enable grad for training, disable for eval/infer
```
[the remark this answers is not in the document]
Comment (zhengzx-nlp) [reply src/main.py]: I agree with you
